airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py

```diff
@@ -382,11 +382,15 @@ def read_records(
                 # Always return an empty generator just in case no records were ever yielded
                 yield from []
         except requests.exceptions.HTTPError as e:
-            status_code = e.response.status_code
-            if status_code == 403:
-                raise RuntimeError(f"Invalid permissions for {self.name}. Please ensure the all scopes are authorized for.")
-            else:
-                raise e
+            raise e
+
+    def parse_response_error_message(self, response):
+        body = response.json()
+        if body["category"] == "MISSING_SCOPES":
+            error = body["errors"][0]
```

Comment:
Should we check that the status is 403 and/or use `.get()` instead to avoid `KeyError`s?

Reply:
good call. updated